play-frontend-hmrc-play-30/src/main/twirl/uk/gov/hmrc/govukfrontend/views/components/GovukCharacterCount.scala.html

```diff
@@ -15,38 +15,36 @@
  *@
 
 @import uk.gov.hmrc.govukfrontend.views.html.components._
+@import scala.collection.immutable.ListMap
 
 @this(govukTextarea: GovukTextarea,
       govukHint: GovukHint)
 
+
 @(params: CharacterCount)
 @import params._
 @require(name.nonEmpty && id.nonEmpty, "name and id should be non-empty")
 
-@attrs = {
-  @maxLength.filter(_ > 0).map {value => data-maxlength="@value"}
-  @threshold.filter(_ > 0).map {value => data-threshold="@value"}
-  @maxWords.filter(_ > 0).map {value => data-maxwords="@value"}
-
-  @govukPluralisedI18nAttributes("characters-under-limit", charactersUnderLimitText)
-  @charactersAtLimitText match {
-      case Some(v) => { data-i18n.characters-at-limit="@{v}" }
-      case None => {}
-  }
-  @govukPluralisedI18nAttributes("characters-over-limit", charactersOverLimitText)
-  @govukPluralisedI18nAttributes("words-under-limit", wordsUnderLimitText)
-  @wordsAtLimitText match {
-      case Some(v) => { data-i18n.words-at-limit="@{v}" }
-      case None => {}
-  }
-  @govukPluralisedI18nAttributes("words-over-limit", wordsOverLimitText)
+@govukPluralisedI18nAttributes(translationKey: String, pluralForms: Option[Map[String, String]]) = @{
+    pluralForms.getOrElse(Map.empty).map { case (key, value) =>
+        s"data-i18n.$translationKey.$key" -> value
+    }
+}
 
-  @if(maxLength.isEmpty && maxWords.isEmpty) {
-      @textareaDescriptionText match {
-          case Some(v) => { data-i18n.textarea-description.other="@{v}" }
-          case None => {}
-      }
-  }
+@dataAttributes = @{
+    val maybeDataAttributes = ListMap(
```

Comment:
do we still need to use a ListMap here? The attrs as string OR map[string, string] is really confusing to work around - looks ok, feels like it's going to be worth seeing if we can do a better comparison when testing against output rendered by govuk stuff, can imagine more an more excluded / flakey tests 😵

Reply:
IIRC `ListMap` fixed some failing unit tests, and yes - it's awful 😅